### DISCUSSION TITLE
Compatible DIGSilent 2009 with CIMTool

### DISCUSSION THREAD

**Discussion by SantiagoRios61970** [cimug-org/CIMTool#62] [General]
Can someone tell me if the exported CIM model files from digsilent can be seen as an instance model in CIMTool or in Enterprise architect. Where can I generate the RDF schema and the XSD schema?
and how would it be done?

**Comment by tviegut** [comment]
Hi @SantiagoRios61970 ,

I will do my best to answer based on limited information and trying to (hopefully) interpret your question correctly.  

First off, I should try to clarify a couple of concepts to ensure we are on the same page in what we are discussing.  For starters your post poses two questions. For the first, it appears that you want some way to "visualize" a CIM/XML file (i.e. "instance model") representing a static network model for a point in time.  Is that correct?  If so I'm assuming then that you're aware that it would not be a one-line diagram in Enterprise Architect but rather what would be known as an "Object Diagram" which represents instances of specific CIM classes and relationships that exist between them.  An "Object Diagram" is different than a "Class Diagram" in EA.  To answer your question directly for that tooling, no there is not a way to generated such a diagram.  You could write a custom script or plugin to do so but nothing that is currently out there.   As for **CIMTool**.  There are no visualization capabilities for generating one-line diagrams (or Object Diagrams for that matter).  So I think that addresses your first question.

On to the second question.  CIMTool is a tool that is used expressly for the purpose you are asking about.  It generates both XSD schemas (and JSON schemas if needed) as well as a variety of RDFS variants.  This generated "profile" (i.e. RDFS artifact) could then provide you with the capabilities of validating your instance data.  Is that the reason you are looking for a way to create the RDFS?  Keep in mind that CIMTool is a tool used by more broadly by the CIM community to generate profiles for data exchanges.  More specifically, it used extensively for creating the official RDFS and/or XSD schemas that accompany IEC International Standards (such as the **IEC 61970-452** standard that defines the profile definitions for excchanging data for static network models on the transmission side of things).  

To create RDFS you must first have a "schema" to import into **CIMTool**.  A schema can be exported from Enterprise Architect and then imported into **CIMTool** before being able to create your profile.  When done in this manner it will be in the form of what is known as an XMI file.  Alternatively, you could use a specific version of the CIM in the form of an Enterprise Architect project file (i.e. `.eap` or `.eapx`) and import that into **CIMTool** as well.  Note that whatever you use it should match the version that your CIM/XML instance file is correlated to.  Again, are you trying to generate an RDFS file for the purpose of validating your specific CIMXML for the specific version of the CIM that it is based upon?  Do you have a different purpose?  The answers to these questions can dictate what you need to do and how (and which) of these applications you reference are to be used to accomplish your goal(s).
 
Hopefully this is helpful.  Please consult the PDF CIMTool presentation ([here](https://github.com/CIMug-org/CIMTool/blob/master/CIMTool_Tutorial.pdf)) that is referenced in the **README** file for this GitHub repo to learn more about what you would need to do to create profiles using **CIMTool**.

Cheers,

Todd

**Reply by SantiagoRios61970** [reply]
Hello, thanks for the information, you can clear my doubts.
Currently I'm develop a project with my University about CIM model and I would like to know how to validate profiles with CIMTool and what can I do with Enterprise Architect?

Also I'm really interested in conformity test and interoperability test, Do you know what kind to tools I can use to have success or where can I to found information about it, some presentation, some pages or pdf's? 

Thanks!

**Reply by Sveino** [reply]
You can check out the information on ENTSO-E in regards to CGMES (IEC 61970-600-series) conformance. https://www.entsoe.eu/data/cim/cim-conformity-and-interoperability/
The site will be updated - so I am happy to get feedback on how to improve the site.

**Reply by tviegut** [reply]
> Hello, thanks for the information, you can clear my doubts. Currently I'm develop a project with my University about CIM model and I would like to know how to validate profiles with CIMTool and what can I do with Enterprise Architect?
> 
> Also I'm really interested in conformity test and interoperability test, Do you know what kind to tools I can use to have success or where can I to found information about it, some presentation, some pages or pdf's?
> 
> Thanks!

@SantiagoRios61970 , @Sveino has pointed you to a great source related to your IOP questions.  I just wanted to follow up with regards to your validation question and **CIMTool**.  That presentation link I gave you should be very helpful in providing guidance on how to use **CIMTool** to validate your instance data.  If you have questions after reviewing that presentation please reach back out so we can assist!

~Todd
